### PR TITLE
Added support for login/register new contacts

### PR DESCRIPTION
**Pull request by shawe**
- Added support on Settings to enable/disable contact logins and new registrations
- Added support on HybridLogin to login a contact
- Added support on LoginToContinue to login a contact
- Added new page to register new contacts
- Ordered json language files, to avoid commit collision

*NOTE 1:* Maybe is useful a reCaptcha on new registrations to filter spam
*NOTE 2:* Maybe is needed a "I don't remember my password" option to restore password
*NOTE 3:* Maybe is needed a "Contact profile" page to allow users to mantain updated here own information or cancel it

### CODE REVIEW

File: Model/Contacto.php [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    public $newPassword2;
+
+    /**
+     * Returns an user by email.
+     *
+     * @param string $email
+     *
+     * @return self|null
+     */
+    public function getByEmail(string $email): ?self
```
Comment (NeoRazorX) [inline Model/Contacto.php]: Unnecesary. You can use loadFromCode('', [new databasewhere...])
And inconsistent with get() method, that return false when fails.

File: Model/Contacto.php [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+     * Returns True if there is no errors on properties values.
+     * It runs inside the save method.
+     *
+     * @return bool
+     */
+    public function test()
+    {
+        $status = parent::test();
+
+        if (isset($this->newPassword, $this->newPassword2) && $this->newPassword !== '' && $this->newPassword2 !== '') {
```
Comment (NeoRazorX) [inline Model/Contacto.php]: Not really necesary. This is a hack solve problems in EditUser.php controller. But in this case you don't use an EditView, so you can simpy test passwords math in controller before call setPassword().

File: View/Macro/PortalTemplate.html.twig [hunk 3]
```diff
@@ -24,16 +24,42 @@
...
-            <a href="#close" class="modal-overlay" aria-label="Close" onclick="return hideModal('loginModal');"></a>
-            <div class="modal-container">
-                <div class="modal-header">
-                    <a href="#close" onclick="return hideModal('loginModal');" class="btn btn-clear float-right" aria-label="Close"></a>
-                    <div class="modal-title h5">{{ i18n.trans('login') }}</div>
-                </div>
-                <div class="modal-body">
-                    <div class="content">
+
+    {% block javascripts %}
```
Comment (NeoRazorX) [inline View/Macro/PortalTemplate.html.twig]: This has no sense. A block inside a macro. Why?

File: View/PortalRegisterMe.html.twig [hunk 4]
```diff
@@ -0,0 +1,66 @@
+{% extends "Master/PortalTemplate.html.twig" %}
+
+{% block body %}
+    <div class="body-section">
+        <br/>
+        <div class="container grid-lg">
+            <div class="columns">
+                <div class="col-12">
```
Comment (NeoRazorX) [inline View/PortalRegisterMe.html.twig]: <div class="column col-12">

File: Model/Contacto.php [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+     * Returns True if there is no errors on properties values.
+     * It runs inside the save method.
+     *
+     * @return bool
+     */
+    public function test()
+    {
+        $status = parent::test();
+
+        if (!isset($this->email) || !filter_var($this->email, FILTER_VALIDATE_EMAIL)) {
```
Comment (NeoRazorX) [inline Model/Contacto.php]: This must go in Core model.

File: Controller/HybridLogin.php [hunk 6]
```diff
@@ -181,4 +194,33 @@ private function twitterLogin()
...
+    {
+        if (AppSettings::get('webportal', 'allowlogincontacts', 'false') === 'false') {
+            return false;
+        }
+
+        $contactEmail = $this->request->request->get('fsContact', '');
+        if ($contactEmail !== '') {
+            $contact = new Contacto();
+            $where = [new DataBaseWhere('email', $contactEmail)];
+            if (!$contact->loadFromCode('', $where)) {
```
Comment (NeoRazorX) [inline Controller/HybridLogin.php]: if( not something ) { ... } else { ... } can be confused. Better to do:
if( something ) { ... } else { ... }

File: Controller/HybridLogin.php [hunk 7]
```diff
@@ -181,4 +194,33 @@ private function twitterLogin()
...
+                $this->miniLog->alert($this->i18n->trans('email-not-registered', [ '%email%' => $contactEmail]));
+            } else {
+                $contactPass = $this->request->request->get('fsContactPass', '');
```
Comment (NeoRazorX) [inline Controller/HybridLogin.php]: If you move $contactPass = to line 210, yo can combine else with the next if.

**Comment by shawe** [comment]
Updated with latest comments on PR

**Pull request merged**